specification/datamigration/resource-manager/Microsoft.DataMigration/preview/2018-07-15-preview/definitions/TasksCommon.json

```diff
@@ -342,6 +451,162 @@
         }
       }
     },
+    "NonSqlMigrationTaskInput": {
+      "type": "object",
+      "description": "Base class for non sql migration task input",
+      "properties": {
+        "targetConnectionInfo": {
+          "$ref": "./Common.json#/definitions/SqlConnectionInfo",
+          "description": "Information for connecting to target"
+        },
+        "targetDatabaseName": {
+          "type": "string",
+          "description": "Target database name"
+        },
+        "projectName": {
+          "type": "string",
+          "description": "Name of the migration project"
+        },
+        "projectLocation": {
+          "type": "string",
+          "description": "An URL that points to the drop location to access project artifacts"
+        },
+        "selectedTables": {
+          "type": "array",
+          "description": "Metadata of the tables selected for migration",
+          "items": {
+            "$ref": "#/definitions/NonSqlDataMigrationTable"
+          }
+        }
+      },
+      "required": [
+        "targetConnectionInfo",
+        "targetDatabaseName",
+        "projectName",
+        "projectLocation",
+        "selectedTables"
+      ]
+    },
+    "NonSqlDataMigrationTable": {
+      "type": "object",
+      "description": "Defines metadata for table to be migrated",
+      "properties": {
+        "sourceName": {
+          "type": "string",
+          "description": "Source table name"
+        }
+      }
+    },
+    "NonSqlMigrationTaskOutput": {
+      "type": "object",
+      "description": "Base class for non sql migration task output",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Result identifier",
+          "readOnly": true
+        },
+        "startedOn": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Migration start time",
+          "readOnly": true
+        },
+        "endedOn": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Migration end time",
+          "readOnly": true
+        },
+        "status": {
+          "$ref": "#/definitions/MigrationStatus",
+          "description": "Current state of migration",
+          "readOnly": true
+        },
+        "dataMigrationTableResults": {
```

Comment:
I know this is not your change, but should this be databaseMigrationTableResults? 
I haven't seen data being used anywhere